File: src/main/java/com/equipment/management/equipment/entity/Equipment.java
package com.equipment.management.equipment.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 设备实体
 */
@Data
@TableName("equipment_base")
public class Equipment implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private String code;

    private String name;

    private String model;

    private Long categoryId;

    private Integer status;

    private String location;

    private String department;

    private LocalDate purchaseDate;

    private String manufacturer;

    private String specification;

    private String qrCode;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;

    /**
     * 分类名称（非数据库字段）
     */
    @TableField(exist = false)
    private String categoryName;
}

File: frontend/src/api/equipment.ts
import request from '@/utils/request'

export interface Equipment {
  id?: number
  code: string
  name: string
  model?: string
  categoryId?: number
  categoryName?: string
  status: number
  location?: string
  department?: string
  purchaseDate?: string
  manufacturer?: string
  specification?: string
  qrCode?: string
}

export interface EquipmentQuery {
  current: number
  size: number
  keyword?: string
  categoryId?: number
  status?: number
}

export function pageEquipments(params: EquipmentQuery) {
  return request.get('/equipment/list', { params })
}

export function getEquipment(id: number) {
  return request.get(`/equipment/${id}`)
}

export function createEquipment(data: Equipment) {
  return request.post('/equipment', data)
}

export function updateEquipment(id: number, data: Equipment) {
  return request.put(`/equipment/${id}`, data)
}

export function deleteEquipment(id: number) {
  return request.delete(`/equipment/${id}`)
}

export function updateEquipmentStatus(id: number, status: number) {
  return request.put(`/equipment/${id}/status`, null, { params: { status } })
}

export function generateQrCode(id: number) {
  return request.get(`/equipment/${id}/qrcode`, { responseType: 'blob' })
}

export function getCategories() {
  return request.get('/equipment/categories')
}
